sksurv/tree/tree.py

```diff
@@ -432,6 +434,13 @@ def predict_cumulative_hazard_function(self, X, check_input=True, return_array=F
         check_is_fitted(self, 'tree_')
         X = self._validate_X_predict(X, check_input, accept_sparse="csr")
 
+        if return_array == "sum":
+            n_samples = X.shape[0]
+            array_of_sums = np.empty(n_samples)
+            for idx in range(0, n_samples):
+                array_of_sums[idx] = self.tree_.predict(X[idx:idx+1])[..., 0].sum()
```

Comment:
This iterates on a one-row-at-a-time basis; an alternative could be to iterate over multiple-rows-at-a-time, somehow, I haven't yet explored how that might work in numpy or so.